### PR TITLE
[IMPROVED] Limit stepdown to system account.

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>

### CODE REVIEW

File: server/jetstream_api.go [hunk 1]
```diff
@@ -2810,6 +2810,12 @@ func (s *Server) jsLeaderStepDownRequest(sub *subscription, c *client, _ *Accoun
 		return
 	}
 
+	// This should only be coming from the System Account.
+	if acc != s.SystemAccount() {
+		s.RateLimitWarnf("JetStream API stepdown request from non-system account: %q user: %q", ci.serviceAccount(), ci.User)
```
Comment (Jarema) [inline server/jetstream_api.go:2815]: Why not return a new error here?

Comment (ripienaar) [reply server/jetstream_api.go:2815]: agree, no response here is good

Comment (Jarema) [reply server/jetstream_api.go:2815]: 👍🏻

**Review by neilalexander — APPROVED** [review]
LGTM

**Review by Jarema — APPROVED** [review]
LGTM

**Pull request merged**